controllers/agents.js

```diff
@@ -7,10 +7,15 @@ async function addNode(req, res) {
         //validate hostId
         const hostId = req.params.hostId
         const osType = req.params.osType;
+        const username = req.body.username;
+        const password = req.body.password;
+        const privateKey = req.body.privateKey;
         if (!hostId || !osType) {
-            return res.status(400).send({ error: 'hostID or osType parameter is required' });
+            return res.status(400).send({ error: 'hostID and osType parameter is required' });
+        }
+        if (!password && !privateKey) {
+            return res.status(400).send({ error: 'either password or privateKey parameter is required' });
```

Comment:
Same here, should just show the problem

Reply:
at least one of them is required. this error shows if none is submitted so it's okay.